testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java

```diff
@@ -172,5 +180,53 @@ public void testLateralViewJsonTupleAs()
                 "SELECT * FROM test_json_tuple_view",
                 queryAssert -> queryAssert.containsOnly(row(3, "Mateusz", "Gajewski", "true", "1000", null, null)));
     }
+
+    @Test(groups = HIVE_VIEWS)
+    public void testUnionViews()
+    {
+        onHive().executeQuery("DROP VIEW IF EXISTS union_all_view");
+        onHive().executeQuery("DROP VIEW IF EXISTS union_view");
+        onHive().executeQuery("DROP VIEW IF EXISTS union_distinct_view");
+
+        Map<String, List<QueryAssert.Row>> expected = new HashMap<>();
+
+        // UNION ALL
+        onHive().executeQuery("CREATE VIEW union_all_view AS\n"
+                + "SELECT n_regionkey FROM nation\n"
+                + "UNION ALL\n"
+                + "SELECT n_regionkey FROM nation\n");
+        expected.put(
+                "union_all_view",
+                // 10 copies of each number from 0 to 4
+                nCopies(10, List.of(row(0), row(1), row(2), row(3), row(4))).stream()
+                        .flatMap(List::stream)
+                        .collect(toList()));
+
+        // UNION DISTINCT
+        onHive().executeQuery("CREATE VIEW union_distinct_view AS\n"
+                + "SELECT n_regionkey FROM nation\n"
+                + "UNION DISTINCT\n"
+                + "SELECT n_regionkey FROM nation\n");
```

Comment:
let's have duplicate rows in one of the sources of the union

UNION DISTINCT should remove all duplicates, regardless whether they come from one source, or multiple, let's verify that